src/browser/keyboard.js

```diff
@@ -492,6 +512,40 @@ export function KeyboardAdapter(bus)
         }
     }
 
+    /**
+     * @param {string} type
+     * @param {?string} data
+     */
+    function handle_inputevent(type, data)
+    {
+        switch(type)
+        {
+            case "insertText":
+                if(data.length > 1)
+                {
+                    // keyboard autocomplete, paste from clipboard
+                    for(var i = 0; i < data.length; i++)
+                    {
+                        keyboard.simulate_char(data[i]);
+                    }
+                }
+                else
+                {
+                    // single character
+                    keyboard.simulate_char(data);
```

Comment:
I think the for loop would handle this case fine (removing the `if`), and would make this code robust to zero-length pastes.